src/authentication/MobileRouter.spec.jsx

```diff
@@ -0,0 +1,227 @@
+import { MobileRouter } from './MobileRouter'
+import React from 'react'
+import { shallow } from 'enzyme'
+import { writeState, writeSecret } from './src/utils/onboarding'
+
+describe('MobileRouter', () => {
+  it('should render the appRoutes when all is well', () => {
+    const app = shallow(
+      <MobileRouter
+        appRoutes={<div />}
+        isAuthenticated={true}
+        isRevoked={false}
+        onboarding={{}}
+        onboardingInformations={{}}
+        history={{}}
+        onAuthenticated={jest.fn()}
+        onLogout={jest.fn()}
+        appIcon={''}
+      />
+    )
+    expect(app).toMatchSnapshot()
+  })
+
+  it('should render the revoked view', () => {
+    const app = shallow(
+      <MobileRouter
+        isAuthenticated={true}
+        isRevoked={true}
+        history={{}}
+        onAuthenticated={jest.fn()}
+        onLogout={jest.fn()}
+        onboarding={{}}
+        onboardingInformations={{}}
+        appIcon={''}
+      />
+    )
+    expect(app).toMatchSnapshot()
+  })
+
+  it('should render the auth screen when the onboarding has not started', () => {
+    const onboardingInformations = {
+      code: null,
+      state: null,
+      cozy_url: null
+    }
+    const app = shallow(
+      <MobileRouter
+        isAuthenticated={false}
+        isRevoked={false}
+        onboarding={{}}
+        onboardingInformations={onboardingInformations}
+        history={{}}
+        onAuthenticated={jest.fn()}
+        onLogout={jest.fn()}
+        appIcon={''}
+      />
+    )
+    expect(app).toMatchSnapshot()
+  })
+
+  it('should render nothing when onboarding', () => {
+    const onboardingInformations = {
+      code: '123',
+      state: 'abc',
+      cozy_url: 'http://lol.com'
+    }
+    const app = shallow(
+      <MobileRouter
+        client={jest.fn()}
+        history={{}}
+        isAuthenticated={false}
+        isRevoked={false}
+        onAuthenticated={jest.fn()}
+        onLogout={jest.fn()}
+        onboarding={{}}
+        onboardingInformations={onboardingInformations}
+        appIcon={''}
+      />
+    )
+    expect(app).toMatchSnapshot()
+  })
+
+  describe('Auto Onboarding', () => {
+    let app,
+      instance,
+      client,
+      onLogout,
+      onAuthenticated,
+      state,
+      code,
+      url,
+      history
+
+    beforeEach(() => {
+      onLogout = jest.fn()
+      onAuthenticated = jest.fn()
+      client = {
+        stackClient: {
+          fetchJSON: jest.fn(),
+          fetch: jest.fn()
+        }
+      }
+      state = '123'
+      code = '111222333'
+      url = 'instance.cozy.test'
+      history = jest.fn()
```

Comment:
you could define all the spies in the `describe` function and just call `jest.clearAllMocks()` or `jest.resetAllMocks()` in `afterEach`?